core/pass.cpp

```diff
@@ -424,3 +440,10 @@ void ClonePass::expr(AST *&ast_)
 
     CompilerPass::expr(ast_);
 }
+
+AST *clone_ast(Allocator &alloc, AST *ast)
```

Comment:
Just so it is easier to call it.  I realized we would probably be calling it quite a lot, and don't want to discourage cloning the subtree, because if the tree becomes a dag it leads to subtle bugs, whereas unnecessarily cloning is pretty harmless (slightly more static memory use).